### ISSUE TITLE
Get access to the official training data

### ISSUE THREAD

**Issue by b-brebion** [zacharycbrown/ssl_baselines_for_biosignal_feature_extraction#1]
Hi!

First of all thanks for your great work. 
I was wondering if the data set used for training was available somewhere because I can't seem to find it. Is there any way to access it in order to train a model without having to adapt the code to another dataset?

Thanks!

**Comment by zacharycbrown** [comment]
Hi!

I'm glad you like it! I will have to ask my advisor if the dataset I am
using has been published, though I suspect it hasn't, unfortunately.

I am currently working on another repo that will hopefully be a little more
flexible and come with a simple dataset for demonstrating the code, but
that will likely be several months out at least.

In the mean time, feel free to ask me any questions you have about
reconfiguring the code and I'll do my best to answer/assist with them. What
ways are you hoping to reconfigure it, if it's ok to ask? Maybe I can
suggest ways to speed things up?

-Zac

**Comment by b-brebion** [comment]
Ok I see, no worries!
At the moment, I'm trying to figure out how to run the code with a public dataset (this one in particular: https://physionet.org/content/challenge-2018/1.0.0/), for an internship project. In this way it allows me to understand a little better how your code works.
By the way, I wanted to know what exactly the `..._TIME.mat` files correspond to, which you get from the `load_time()` function. Because in the public dataset I use, I only have the signals files.

Thanks in advance!

**Comment by zacharycbrown** [comment]
Great question!

I had to review that code rather closely because I wasn't the one who wrote
the bulk of that code (data_utils.py is an adaptation of code pulled from
Jason Stranne's github.com/jstranne/mouse_self_supervision). Here's my
current understanding (if I prove myself wrong later, I'll try to remember
to let you know):
    The ..._TIME.mat files are used to help with generating trinary labels
for the time series. If I understand correctly, these files are structured
as [label_1_start_time, label_1_duration, label_2_start_time,
label_2_duration] (with the entries being measured in seconds) - any region
of the recording that falls outside of the regions defined by these entries
receives a third label_0. The key lines to look for in data_utils.py where
the _TIME.mat files are used are lines 97 as well as 130-132. On line 97,
..._TIME.mat files are being used to 'crop' the time series to extract
regions that we are fairly confident will exhibit the label_0, label_1, and
label_2 behaviors/features. On lines 130-132, the corresponding label
vector for the 'cropped' time series is being generated (and I believe will
end up looking like [0,...,0,1,...,1,2,...,2].

Does that answer clarify things a little better? Looking at it more
closely, it's very clear that some of the documentation in that file ought
to be fixed (one reason why I'm building out the other repository is to
make things clearer) ...

-Zac

**Comment by b-brebion** [comment]
Thanks for your explanation! I understand more clearly the use of these files in the code.
So fundamentally, these labels just serve to keep certain parts of the signals that have been chosen beforehand?
And since I don't have these files in the dataset I use, can I just go without them? Or will it be a problem when training, for example?

**Comment by zacharycbrown** [comment]
The _TIME.mat files are only used for generating the labels corresponding
to each time step for supervised training (in the downstream task) - as
long as you have labels for the downstream training, I don't believe you
need the _TIME.mat files, just ensure that the train_loader has the same
structure as is used in the downstream training loop.

**Comment by b-brebion** [comment]
Alright, thank you very much!
I will come back to you if I have more questions in the future 😄

**Comment by zacharycbrown** [comment]
No problem, any time!
